desc/coils.py

```diff
@@ -1198,8 +1214,29 @@ def from_symmetry(cls, coils, NFP=1, sym=False):
             rotated_coils = coils.copy()
             rotated_coils.rotate(axis=[0, 0, 1], angle=2 * jnp.pi * k / NFP)
             coilset += rotated_coils
+        coilset_to_return = cls(*coilset)
+        if check_coil_coil_dists:
+            grid = LinearGrid(N=100)
+            min_dists = coilset_to_return.compute_minimum_intercoil_distances(grid=grid)
+            intersection = np.any(min_dists < np.finfo(min_dists.dtype).eps)
+            near_intersection = np.any(min_dists < 1e-2)
```

Comment:
For a discretized coil I think the only way they'd be actually touching is if they're almost literally the same coil in the same position, so it might be better to just lump everything into near intersections

Also, for near intersection it's probably good to scale things by some characteristic size and account for the finite grid spacing. A simple heuristic could be "if the closest point to zeta=x on coil 0 is on a different coil"?

Reply:
True, I will add tmrw but if anuone wants to beat me to it go ahead

Reply:
compute dist matrix and check if argmin along each axis is different than the the inds corresponding to that coil

Reply:
I think what you have now also works fine. Assuming the objective is normalized, a sensible default tol would be 1/number of grid poiints, I think that would be pretty similar to what we talked about but probably easier to implement

Reply:
I implemented your suggestion, not really sure which is better tbh. My original implementation was a bit cleaner but did rely on a tol.

It is not clear to me which way is more robust either

Reply:
For now my test for this passes fine, so I think I am good with keeping this implementation for now. We could also do it where if you pass in `tol` it will use my original implementation, and if you dont it defaults to `"auto"` which uses this way of checking based off of which coil pt is closest